bandersnatch_vrfs/Cargo.toml

```diff
@@ -30,12 +30,29 @@ ark-bls12-381 = { version = "0.4", default-features = false, features = [ "curve
 sha2 = { version = "0.10", default-features = false }
 rand_chacha = { version = "0.3.1", default-features = false }
 
-[dev-dependencies]
+# Substrate curves are optional and gated by the 'substrate-curves' feature
+sp-ed-on-bls12-381-bandersnatch = { git = "https://github.com/paritytech/substrate-curves", default-features = false, optional = true }
+sp-bls12-381 = { git = "https://github.com/paritytech/substrate-curves", default-features = false, optional = true }
 
 
 [features]
-default = ["getrandom"]  #  "std",
+default = ["std"]
+std = [
+  "getrandom",
+  "ring/std",
+  "ark-std/std",
+  "ark-ff/std",
+  "ark-ec/std",
+  "ark-serialize/std",
+  "sp-ed-on-bls12-381-bandersnatch?/std",
+  "sp-bls12-381?/std",
+]
 getrandom = ["dleq_vrf/getrandom"]
-std = ["getrandom", "ring/std", "ark-std/std", "ark-ff/std", "ark-ec/std", "ark-serialize/std"]
-# parallel = ["std", "ring/parallel"]  #  ["std", "rayon", "ark-std/parallel", "ark-ff/parallel", "ark-ec/parallel", "ark-poly/parallel"]
 print-trace = ["ark-std/print-trace"]
+# Substrate curves allows to offload computationally heavy tasks to Substrate host functions.
+# Mostly useful in Substrate development context when targeting wasm32 architecture.
+substrate-curves = [
+  "sp-ed-on-bls12-381-bandersnatch",
+  "sp-bls12-381",
+]
```

Comment:
I forget, but these curves should only work from the substrate runtime, no?  Or they fall back to arkworks' curve?

What does `sp-*` mean?    I'd thought we named these curves `sub-ark-*` because only crates in the substrate tree use `sp-*`.

Reply:
> I forget, but I'd thought these curves should only work within the substrate runtime, no? Or they fall back to arkworks' curve?

These curves are in practice used only from the runtime and are exposed by the Substrate's crate `sp-crypto-ec-utils`.
([here](https://github.com/paritytech/polkadot-sdk/blob/cb944dc54882222453071fc99f2e5014c51823ef/substrate/primitives/crypto/ec-utils/src/lib.rs#L37-L38)).

This runtime interface is used to construct the interface of a bunch of host functions which, by definition, are mainly used from the runtime BUT can also be called from the native code as well.  In practice nothing bad happens if native code calls a crypto primitive passing through this path instead calling it directly via `sp-core`, but is also useless as it takes a not required detour. 

> What does sp-* mean? 

`sp-*` is a prefix used for substrate primitives.

> I'd thought we named these curves sub-ark-* because only crates in the substrate tree use sp-*.

There are two crates families (which at the moment are hosted in separate repos)

- [`sp-ark-*`](https://github.com/paritytech/ark-substrate/) : here is where ark configs are defined to eventually jump into externally provided hooks for the computationally heavy stuff. Here we don't provide any implementation and indeed is not related to substrate at all. Indeed I want to rename the crates to something like `ark-...-ext` following the same name of ark upstream. As I said these crates are generic and not related to Substrate in any way.

- [`sp`](https://github.com/paritytech/substrate-curves): here is where the `sp-ark` crates are specialized to use the Substrate host functions. So the hooks are constructed to jump into substrate `ec-utils` (i.e. host functions interface I was previosly talking about). Maybe these crates will be merged directly into substrate as they are strictly bound to substrate and can't be used for any other thing. IMO doesn't make much sense to keep them separated as just introduce more maintanance work to keep them up to date.

Reply:
> sp-* is a prefix used for substrate primitives.

Yes, but should they play a specific role in substrate?  Or can we use them from "somewhat" outside substrate like this?  I'm just asking if the crates are named correctly according to the conventions.